src/gui/menu_filesystem.cpp

```diff
@@ -62,7 +65,13 @@ FileSystemMenu::refresh_items()
   m_files.clear();
   m_directory = FileSystem::normalize(m_directory);
 
-  add_label(m_directory);
+  // Make sure label doesn't get too long.
+  std::string title = m_directory;
+  const bool title_large = title.size() > s_title_max_chars;
+  while (title.size() > s_title_max_chars) title = title.substr(title.size() - s_title_max_chars);
```

Comment:
Maybe it would be better to do:
```cpp
if(title.size() > s_title_max_chars)
  title = title.substr(s_title_max_chars-3)+"...";
```
? It will show beginning of title instead.